utils/spdx/src/main/kotlin/parser/SpdxExpressionParser.kt

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (C) 2024 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils.spdx.parser
+
+import org.ossreviewtoolkit.utils.common.nextOrNull
+import org.ossreviewtoolkit.utils.spdx.SpdxCompoundExpression
+import org.ossreviewtoolkit.utils.spdx.SpdxExpression
+import org.ossreviewtoolkit.utils.spdx.SpdxLicenseIdExpression
+import org.ossreviewtoolkit.utils.spdx.SpdxLicenseReferenceExpression
+import org.ossreviewtoolkit.utils.spdx.SpdxLicenseWithExceptionExpression
+import org.ossreviewtoolkit.utils.spdx.SpdxOperator
+
+/**
+ * A parser for SPDX expressions. It consumes a sequence of [Token]s and produces an [SpdxExpression].
+ *
+ * This parser implements the grammar defined in the
+ * [SPDX specification](https://spdx.github.io/spdx-spec/v2.2.2/SPDX-license-expressions/):
+ *
+ * ```
+ * license-expression   -> simple-expression | compound-expression
+ * compound-expression  -> simple-expression |
+ *                         simple-expression "WITH" license-exception-id |
+ *                         compound-expression "AND" compound-expression |
+ *                         compound-expression "OR" compound-expression |
+ *                         "(" compound-expression ")" )
```

Comment:
Line 42 has a closing brace without a corresponding opening brace.

Can the braces be dropped?